check_runner/__init__.py

```diff
@@ -0,0 +1,153 @@
+the executed program returned an incompatible exit code and a `UNKNOWN` status if the service status
+is uncertain.
+Note: if this script is used as Nagios Plugin then the `debug` option must be disabled.
+"""
+
+import argparse
+import logging
+import subprocess
+
+import time
+import cachetclient.cachet as cachet
+
+from . import settings
+from monitoring_scripts import nagios_common
+
+parser = argparse.ArgumentParser(
+    prog='cnto-check-runner',
+    description=__doc__,
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter
+)
+parser.add_argument(
+    'script',
+    help='path to executable script which monitors a service and returns its status with a '
+         'Nagios-compatible standard'
+)
+parser.add_argument(
+    'component_id',
+    help='numerical id of Cachet\'s component',
+    type=int
+)
+parser.add_argument(
+    'config_file',
+    help='path to configuration file, see config.ini.dist file for example'
+)
+parser.add_argument(
+    '--retries',
+    help='number of attempts to run if the first fails before a hard failure is actually filed to '
+         'Cachet',
+    type=int,
+    default=5
+)
+parser.add_argument(
+    '--interval',
+    help='time (in seconds) between each attempt',
+    type=float,
+    default=0.5
+)
+parser.add_argument(
+    '--script-args',
+    help='arguments passed to the monitoring script',
+    nargs=argparse.REMAINDER,
+    default=[]
+)
+parser.add_argument(
+    '--debug',
+    help='display debug messages',
+    default=False,
+    action='store_true'
+)
+
+codes_mapping = {
+    nagios_common.Codes.OK: 1,
+    nagios_common.Codes.WARNING: 4,
+    nagios_common.Codes.CRITICAL: 4,
+    nagios_common.Codes.UNKNOWN: None
+}
+
+
+def main(script, component_id, config_file, script_args=None, retries=5, interval=0.5, debug=False):
+    """Script execution"""
+
+    if script_args is None:
+        script_args = []
+
+    logging.basicConfig(level=logging.INFO)
+    logger = logging.getLogger(__name__)
+
+    if debug: # pragma: no cover
+        logger.setLevel(logging.DEBUG)
+        logger.info('debug logging enabled')
+
+    # Read Cachet API parameters from config file
+    config_parser = settings.RunnerConfigParser()
+    config_parser.read(config_file)
+    base_url = config_parser.get('Cachet', 'base-url', fallback=None)
+    api_key = config_parser.get('Cachet', 'api-key', fallback=None)
+    logger.info('Config params: base-url = %s, api-key = %s', base_url, api_key)
+    if not api_key or not base_url:
+        logger.critical('one or both mandatory config parameters are missing')
+        raise ValueError('Invalid config file, use config.ini.dist file as reference')
+
+    # Run monitoring script
+    script_args.insert(0, script)
+    logger.debug('invocation arguments %s', script_args)
+    script_return_code = None
+    remaining_attempts = retries
+    while script_return_code != nagios_common.Codes.OK and remaining_attempts >= 0:
```

Comment:
doing a `for retry_num in range(retries, 0, -1):` then `break` if `script_return_code` is not `OK` would feel more readable, what do you think ?

Reply:
I was taught to avoid `break` as much as possible and I think the `while` loop fits in well, although I realise the solution you propose is more "pythonic", I'll change it.